internal/controller/cluster_monitoring_controller.go

```diff
@@ -112,6 +115,30 @@ func SetupClusterMonitoringReconciler(mgr manager.Manager, conf config.Config) e
 func (r *ClusterMonitoringReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.Cluster{}).
+		// Reconcile all clusters when the grafana organizations have changed
+		Watches(&v1alpha1.GrafanaOrganization{},
```

Comment:
Why do we need this ? What's the relation with rules loading ?

Reply:
This is needed to ensure cluster alloy metrics configs are updated (the remote write sections) when tenants and specifically the tenants field is updated. I initially needed it when alloy-rules was created by the cluster controller but it is actually useful

Reply:
You mean here https://github.com/giantswarm/observability-operator/pull/330#discussion_r2026803821 ? 😀